src/languages/swift.js

```diff
@@ -1,65 +1,66 @@
-        'bridgeToObjectiveCUnconditional c compactMap contains count countElements countLeadingZeros ' +
-        'debugPrint debugPrintln distance dropFirst dropLast dump encodeBitsAsWords ' +
-        'enumerate equal fatalError filter find getBridgedObjectiveCType getVaList ' +
-        'indices insertionSort isBridgedToObjectiveC isBridgedVerbatimToObjectiveC ' +
-        'isUniquelyReferenced isUniquelyReferencedNonObjC join lazy lexicographicalCompare ' +
-        'map max maxElement min minElement numericCast overlaps partition posix ' +
-        'precondition preconditionFailure print println quickSort readLine reduce reflect ' +
-        'reinterpretCast reverse roundUpToAlignment sizeof sizeofValue sort split ' +
-        'startsWith stride strideof strideofValue swap toString transcode ' +
-        'underestimateCount unsafeAddressOf unsafeBitCast unsafeDowncast unsafeUnwrap ' +
-        'unsafeReflect withExtendedLifetime withObjectAtPlusZero withUnsafePointer ' +
-        'withUnsafePointerToObject withUnsafeMutablePointer withUnsafeMutablePointers ' +
-        'withUnsafePointer withUnsafePointers withVaList zip'
-    };
+  // https://docs.swift.org/swift-book/ReferenceManual/LexicalStructure.html#ID413
+  // https://docs.swift.org/swift-book/ReferenceManual/zzSummaryOfTheGrammar.html
+  const DOT_KEYWORD = {
+    begin: concat(/\./, either(...Swift.dotKeywords, ...Swift.optionalDotKeywords)),
+    returnBegin: true,
+    contains: [
+      {
+        begin: /\./
+      },
+      {
+        className: 'keyword',
+        begin: either(...Swift.dotKeywords, ...Swift.optionalDotKeywords)
+      }
+    ]
+  };
+  const KEYWORD_GUARD = {
+    // Consume .keyword to prevent highlighting properties and methods as keywords.
+    begin: concat(/\./, either(...Swift.keywords)),
+    relevance: 0,
+  };
+  const KEYWORD = {
+    className: 'keyword',
+    variants: [
+      { begin: either(...Swift.keywords, ...Swift.optionalDotKeywords) },
+      { begin: concat(/#/, either(...Swift.numberKeywords)) }
+    ]
+  };
+  const KEYWORDS = [DOT_KEYWORD, KEYWORD_GUARD, KEYWORD];
 
-  const TYPE = {
-    className: 'type',
-    begin: '\\b[A-Z][\\w\u00C0-\u02B8\']*',
-    relevance: 0
+  // https://github.com/apple/swift/tree/main/stdlib/public/core
+  const BUILT_IN_GUARD = {
+    // Consume .built_in to prevent highlighting properties and methods.
+    begin: concat(/\./, either(...Swift.builtIns)),
+    relevance: 0,
   };
-  // slightly more special to swift
-  const OPTIONAL_USING_TYPE = {
-    className: 'type',
-    begin: '\\b[A-Z][\\w\u00C0-\u02B8\']*[!?]'
+  const BUILT_IN = {
+    className: 'built_in',
+    begin: concat(/\b/, either(...Swift.builtIns), /(?=\()/)
+  };
+  const BUILT_INS = [BUILT_IN_GUARD, BUILT_IN];
+
+  // https://docs.swift.org/swift-book/ReferenceManual/LexicalStructure.html#ID418
+  const OPERATOR = {
+    className: 'operator',
+    variants: [
```

Comment:
```suggestion
    className: 'operator',
    relevance: 0,
    variants: [
```

Operators are not given relevance, too common and not in enough grammars yet.

Reply:
Many of your auto-detect issues are probably right here + plus the double/triple relevance issues.

Reply:
Done (in the next commit).